src/status_im/ui/screens/wallet_connect/session_proposal/views.cljs

```diff
@@ -0,0 +1,215 @@
+    [quo/text {:size :small} (i18n/label :t/select-account)]
+    [list/flat-list {:data        accounts
+                     :key-fn      :address
+                     :render-fn   render-account
+                     :render-data {:selected-account selected-account
+                                   :on-select on-select}
+                     :horizontal  true
+                     :shows-horizontal-scroll-indicator false
+                     :extraData @selected-account
+                     :style styles/account-selector-list}]]])
+
+(defn account-picker [accounts selected-account {:keys [on-press on-select]}]
+  (if (> (count accounts) 1)
+    [react/view {:style styles/account-selector-wrapper}
+     [account-selector accounts selected-account on-select]]
+    [react/touchable-opacity {:style styles/single-account-container}
+     [toolbar-selection {:text (:name @selected-account)
+                         :background-color (:color @selected-account)
+                         :on-press on-press}]]))
+
+(defview success-sheet-view [{:keys [topic]}]
+  (letsubs [visible-accounts @(re-frame/subscribe [:visible-accounts-without-watch-only])
+            dapps-account [:dapps-account]
+            sessions [:wallet-connect/sessions]
+            managed-session [:wallet-connect/session-managed]]
+    (let [{:keys [peer state] :as session} (first (filter #(= (:topic %) topic) sessions))
+          {:keys [accounts]} state
+          {:keys [metadata]} peer
+          {:keys [name icons]} metadata
+          icon-uri (when (and icons (> (count icons) 0)) (first icons))
+          address (last (string/split (first accounts) #":"))
+          account (first (filter #(= (:address %) address) visible-accounts))
+          selected-account (reagent/atom account)]
+      [react/view (styles/acc-sheet)
+       [react/view (styles/proposal-sheet-container)
+        [react/view (styles/proposal-sheet-header)
+         [quo/text {:weight :bold
+                    :size   :large}
+          (i18n/label :t/connection-request)]]
+        [react/image {:style (styles/dapp-logo)
+                      :source {:uri icon-uri}}]
+        [react/view styles/sheet-body-container
+         [react/view styles/proposal-title-container
+          [quo/text {:weight :bold
+                     :size   :large}
+           name]
+          [quo/text {:weight :regular
+                     :size   :large
+                     :style  styles/proposal-title}
+           (i18n/label :t/connected)]]]
+        [account-picker
+         (vector dapps-account)
+         selected-account
+         {:on-press #(do
+                       (re-frame/dispatch [:wallet-connect/manage-app session])
+                       (reset! show-account-selector? true))}]
+        [quo/text {:weight :regular
+                   :color :secondary
+                   :style  styles/message-title}
+         (i18n/label :t/manage-connections)]
+        [react/view (styles/footer)
```

Comment:
why so many nested views?

Reply:
Not specific reason in this case, just refactored it